src/sqlfluff/core/dialects/ansi_keywords.py

```diff
@@ -16,6 +16,9 @@
 UNION
 IGNORE
 RESPECT
+PARTITION
```

Comment:
Changed these to reserved keywords in order to address a tricky parsing issue. From the BigQuery [docs](https://cloud.google.com/bigquery/docs/reference/standard-sql/analytic-function-concepts#def_window_spec), this is the syntax for a window specification:

```
window_specification:
  [ named_window ]
  [ PARTITION BY partition_expression [, ...] ]
  [ ORDER BY expression [ { ASC | DESC } ] [, ...] ]
  [ window_frame_clause ]
```

The leading (but optional) `named_window` makes parsing tricky. Until I made this change, a simple specification without a name would not parse correctly, e.g. in `PARTITION BY category`, the parser would think `PARTITION` was a `named_window`.

Are there any concerns with this change? If so, any thoughts on alternative designs?

Reply:
I think it's fair to have "PARTITION" as a reserved keyword, another way I could see, would be a negative match regex that excludes "partition" from valid window names

Reply:
I think I'll stay with the above approach. We could always revisit it later, but my intuition for now is that the simplicity of having fewer special-case lexing patterns is a good thing.